File: _submissions/18UCS045_charan/DAY1/prime sum.cpp
vector<bool> isPrime(long n){
    vector<bool> prime(n+1,true);
    prime[0] = prime[1] = false;
    for(long i=2;i*i<=n;i++){
        if(prime[i]){
            for(long j=i*i;j<=n;j+=i){
                prime[j] = false;
            }
        }
    }
    return prime;
}

vector<int> Solution::primesum(int A) {
    vector<int> ans;
    vector<bool> prime = isPrime(A);
    for(int i=2;i<A;i++){
        if(prime[i] and prime[A-i]){
            ans.push_back(i);
            ans.push_back(A-i);
            return ans;
        }
    }
    return ans;

File: _submissions/19UCS159_Tharun/Day1/primeSum.java
public class Solution {
  public int[] primesum(int A) {
    boolean[] isPrime = new boolean[A + 1];
    Arrays.fill(isPrime, true);
    isPrime[0] = isPrime[1] = false;
    for (int i = 2; i * i <= A; i++) {
      if (isPrime[i]) {
        for (int j = i * i; j <= A; j += i) {
          isPrime[j] = false;
        }
      }
    }
    for (int p = 2; p <= A / 2; p++) {
      if (isPrime[p] && isPrime[A - p]) {
        return new int[] { p, A - p };
      }
    }
    return null;
  }
}
